File: ch01-solid-principles/s02-open-close/ex02-switch-clean/src/main/java/com/example/strategies/DiscountStrategyMap.java
package com.example.strategies;

import com.example.IDiscountStrategy;

import java.util.Map;
import java.util.HashMap;

// tag::source[]
public class DiscountStrategyMap {
    private static Map<Integer, IDiscountStrategy> strategyMap = createStrategies();

    private static Map<Integer, IDiscountStrategy> createStrategies() {
        Map<Integer, IDiscountStrategy> map = new HashMap<>();
        IDiscountStrategy janFebStrategy = new JanFebDiscountStrategy();
        IDiscountStrategy augStrategy = new AugDiscountStrategy();
        IDiscountStrategy otherStrategy = new OtherDiscountStrategy();
        map.put(1, janFebStrategy);
        map.put(2, janFebStrategy);
        map.put(3, otherStrategy);
        map.put(4, otherStrategy);
        map.put(5, otherStrategy);
        map.put(6, otherStrategy);
        map.put(7, otherStrategy);
        map.put(8, augStrategy);
        map.put(9, otherStrategy);
        map.put(10, otherStrategy);
        map.put(11, otherStrategy);
        map.put(12, otherStrategy);

        return map;
    }

    public static IDiscountStrategy get(int month) {
        return strategyMap.get(month);
    }
}
// end::source[]

File: ch01-solid-principles/s02-open-close/ex02-switch-clean/src/main/C#/SwitchClean/DiscountStrategyMap.cs
using SwitchClean.strategies;
using System.Collections.Generic;

namespace SwitchClean
{
    public class DiscountStrategyMap
    {
        public static Dictionary<int, IDiscountStrategy> strategyMap = new Dictionary<int, IDiscountStrategy>()
        {
            {1, new JanFebDiscountStrategy()},
            {2,new JanFebDiscountStrategy()},
            {3, new OtherDiscountStrategy()},
            {4, new OtherDiscountStrategy()},
            {5, new OtherDiscountStrategy()},
            {6, new OtherDiscountStrategy()},
            {7, new OtherDiscountStrategy()},
            {8, new AugDiscountStrategy()},
            {9, new OtherDiscountStrategy()},
            {10, new OtherDiscountStrategy()},
            {11, new OtherDiscountStrategy()},
            {12, new OtherDiscountStrategy()}
        };
    }
}
